### PR TITLE
🏄🏼‍♂️ [Feature] Add push notifications 

### PR DESCRIPTION
**Pull request by kcw-grunt**
# Overview
To reach and engage users we are adding push notifications and  email signup [here](https://github.com/litecoin-foundation/litewallet-android/issues/159).

## Work done

1. Added Push Notifications dependencies
2. Added Advertsing ID and its dependencies

### CODE REVIEW

**Review by josikie — APPROVED** [review]
Wow, thank you! @kcw-grunt

File: app/src/main/java/com/breadwallet/BreadApp.java [hunk 1]
```diff
@@ -183,25 +149,35 @@ public void run() {
...
+
+            String adInfoString = adInfo.getId();
+            String generalAndroidInterest = "general-android";
+            String debugGeneralAndroidInterest = "debug-general-android";
+            String pusherInstanceID = Utils.fetchPartnerKey(this, PartnerNames.PUSHERSTAGING);
+
+            // setup Push Notifications
+            //This worked had to add the iid dep https://github.com/mixpanel/mixpanel-android/issues/744
+            PushNotifications.start(getApplicationContext(), pusherInstanceID);
+            PushNotifications.addDeviceInterest(generalAndroidInterest);
```
Comment (josikie) [inline app/src/main/java/com/breadwallet/BreadApp.java:169]: What is addDeviceInterest?

**Review by josikie — APPROVED** [review]
Wow, thank you! @kcw-grunt

File: app/src/main/java/com/breadwallet/BreadApp.java (hunk 1 above)
Comment (kcw-grunt) [reply app/src/main/java/com/breadwallet/BreadApp.java:169]: these are a way for pusher to send to different groups. For instance we might want to push to Spanish speaking people. The push would be in Spanish and we might add the interest `español-android-interest` & `español-ios-interest` and just send pushes to them

Comment (josikie) [reply app/src/main/java/com/breadwallet/BreadApp.java:169]: thanks for the explanation

**Pull request merged**